docs/conf.py

```diff
@@ -1,131 +1,70 @@
+from pkg_resources import get_distribution, DistributionNotFound
+
+# -- Check for dependencies ----------------------------------------------------
+
+doc_requires = get_distribution("sunpy").requires(extras=("docs",))
+missing_requirements = []
+for requirement in doc_requires:
+    try:
+        get_distribution(requirement)
+    except Exception as e:
+        missing_requirements.append(requirement.name)
+if missing_requirements:
+    print(
+        f"The {', '.join(missing_requirements)} package(s) could not be found and "
+        "is needed to build the documentation, please install the 'docs' requirements.")
     sys.exit(1)
 
-try:
-    import sphinx_gallery
-    if on_rtd and os.environ.get('READTHEDOCS_PROJECT').lower() != 'sunpy':
-        # Gallery takes too long on RTD to build unless you have extra build time.
-        has_sphinx_gallery = False
-    else:
-        has_sphinx_gallery = True
-except ImportError:
-    has_sphinx_gallery = False
+# -- Read the Docs Specific Configuration --------------------------------------
 
+# This needs to be done before sunpy is imported
+on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
 if on_rtd:
     os.environ['SUNPY_CONFIGDIR'] = '/home/docs/'
     os.environ['HOME'] = '/home/docs/'
     os.environ['LANG'] = 'C'
     os.environ['LC_ALL'] = 'C'
 
-try:
-    import zeep
-except ImportError as e:
-    raise Exception(e, 'ERROR: zeep could not be imported. Building the documentation requires '
-                    'the "zeep" package to be installed')
-
-try:
-    import skimage
-except ImportError as e:
-    raise Exception(e, 'ERROR: skimage could not be imported. Building the documentation requires '
-                    'the "scikit-image" package to be installed')
+# -- Non stdlib imports --------------------------------------------------------
 
-try:
-    import drms
-except ImportError as e:
-    raise Exception(e, 'ERROR: drms could not be imported. Building the documentation requires '
-                    'the "drms" package to be installed')
+import ruamel.yaml as yaml
+from sphinx_gallery.sorting import ExplicitOrder
+from sphinx_gallery.sorting import ExampleTitleSortKey
 
-try:
-    import glymur
-except ImportError as e:
-    raise Exception(e, 'ERROR: glymur could not be imported. Building the documentation requires '
-                    'the "glymur" package to be installed')
+import sunpy
+from sunpy import __version__
 
-try:
-    import sqlalchemy
-except ImportError as e:
-    raise Exception(e, 'ERROR: sqlalchemy could not be imported. Building the documentation requires '
-                    'the "sqlalchemy" package to be installed')
+# -- Project information -------------------------------------------------------
 
-try:
-    import astroquery
-except ImportError as e:
-    raise Exception(e, 'ERROR: astroquery could not be imported. Building the documentation requires '
-                    'the "astroquery" package to be installed')
+project = 'SunPy'
```

Comment:
```suggestion
project = 'sunpy'
```
Since it's the package and not the project?

Reply:
While you might be right, this would make me sad (and also this is how it is already).